### PR TITLE
Fix propagation of device and dtype properties in Lite modules

### LINKED ISSUE #10556: LightningLite not updating DeviceDtypeModuleMixin correctly
## 🐛 Bug

When using LightningLite and transferring the `_LiteModule` to cpu, attributes of `DeviceDtypeModule` are not updated.

### To Reproduce

```python

class SomeDummy(DeviceDtypeModuleMixin):
    def __init__(self):
        super().__init__()
        self.a = torch.nn.Linear(1,1)

class MyClass(LightningLite):
    def run(self):
        model = SomeDummy()
        model, optimiser = self.setup(model, torch.optim.Adam(model.parameters()))

        #  do some stuff
        
        # now clean up gpu memory for later stages
        model.cpu()
        assert str(model.module.device) == 'cpu'

MyClass(accelerator='gpu', devices=1).run()
```
      
### Expected behavior

model.module.device should be Cpu

### Additional context

Could probably be solved by using `DeviceDtypeModuleMixin` as base class for the `_LiteModule` since this is an issue with the `to` function only calling `_apply` on all child tensors instead of calling `.to` on every child module.

cc @carmocca @justusschock @awaelchli

### PR DESCRIPTION
**Pull request by awaelchli**
## What does this PR do?

Fixes #10556

## Before submitting
- [x] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes? (if necessary)
- [x] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- [x] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?
I made sure I had fun coding 🙃

Part of #1 (it's a lie, this is just here to avoid noisy GitHub bot)

cc @borda

### CODE REVIEW

**Review by tchaton — APPROVED** [review]
LGTM !

File: pytorch_lightning/lite/wrappers.py [hunk 1]
```diff
@@ -64,7 +65,7 @@ def step(self, closure: Optional[Callable] = None) -> None:
         )
 
 
-class _LiteModule(nn.Module):
+class _LiteModule(DeviceDtypeModuleMixin):
     def __init__(self, module: nn.Module, precision_plugin: PrecisionPlugin) -> None:
         """The LiteModule is a thin wrapper around the :class:`torch.nn.Module` and handles precision / autocast
```
Comment (carmocca) [inline pytorch_lightning/lite/wrappers.py:70]: Update the docstring

**Review by carmocca — APPROVED** [review]

File: pytorch_lightning/lite/wrappers.py (hunk 1 above)
Comment (awaelchli) [reply pytorch_lightning/lite/wrappers.py:70]: What do you suggest? The current sentence is still correct.

**Review by rohitgr7 — APPROVED** [review]

**Review by justusschock — APPROVED** [review]
